src/de/grobox/liberario/fragments/SettingsFragment.java

```diff
@@ -92,30 +93,30 @@ public boolean onPreferenceClick(Preference preference) {
 		// Fill in current home location if available
 		if(network != null) setHome(null);
 
-		home = findPreference("pref_key_create_quickhome_shortcut");
-		home.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
+		quickhome = findPreference("pref_key_create_quickhome_shortcut");
+		quickhome.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
 			public boolean onPreferenceClick(Preference preference) {
 				// create launcher shortcut
+				Intent shortcutIntent = new Intent(DirectionsFragment.TAG, Uri.EMPTY, getContext(), MainActivity.class);
+				shortcutIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+				shortcutIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+				shortcutIntent.putExtra("special", DirectionsFragment.TASK_BRING_ME_HOME);
+
+				Intent addIntent = new Intent();
+				addIntent.putExtra(Intent.EXTRA_SHORTCUT_INTENT, shortcutIntent);
+				addIntent.putExtra(Intent.EXTRA_SHORTCUT_NAME, getString(R.string.widget_name_quickhome));
+				addIntent.putExtra(Intent.EXTRA_SHORTCUT_ICON_RESOURCE, Intent.ShortcutIconResource.fromContext(getContext(), R.drawable.ic_quickhome_widget));
+				addIntent.putExtra("duplicate", false);
+				addIntent.setAction("com.android.launcher.action.INSTALL_SHORTCUT");
+				getContext().sendBroadcast(addIntent);
 				int currentapiVersion = Build.VERSION.SDK_INT;
 				if(currentapiVersion >= Build.VERSION_CODES.KITKAT) {
-					// the INSTALL_SHORTCUT permission is not available below API 19
-					Intent shortcutIntent = new Intent(DirectionsFragment.TAG, Uri.EMPTY, getContext(), MainActivity.class);
-					shortcutIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-					shortcutIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
-					shortcutIntent.putExtra("special", DirectionsFragment.taskBringMeHome);
-
-					Intent addIntent = new Intent();
-					addIntent.putExtra(Intent.EXTRA_SHORTCUT_INTENT, shortcutIntent);
-					addIntent.putExtra(Intent.EXTRA_SHORTCUT_NAME, getString(R.string.widget_name_quickhome));
-					addIntent.putExtra(Intent.EXTRA_SHORTCUT_ICON_RESOURCE, Intent.ShortcutIconResource.fromContext(getContext(), R.drawable.ic_quickhome_widget));
-					addIntent.putExtra("duplicate", false);
-					addIntent.setAction("com.android.launcher.action.INSTALL_SHORTCUT");
-					getContext().sendBroadcast(addIntent);
 					Toast.makeText(getContext(), getString(R.string.toast_create_quickhome_shortcut), Toast.LENGTH_SHORT).show();
 				} else {
-					Toast.makeText(getContext(), getString(R.string.error_cannot_create_quickhome_shortcut), Toast.LENGTH_SHORT).show();
+					// the INSTALL_SHORTCUT permission is part of the launcher below API 19 and
+					// therefore might not be supported
+					Toast.makeText(getContext(), getString(R.string.toast_create_quickhome_old_API), Toast.LENGTH_SHORT).show();
```

Comment:
I liked it more this way, because the user can try if it works (if not, this is just sending an Intent).
If you prefer I can also remove the permission below API 19.